File: bus-booking-frontend/src/services/routeService.js
import api from "../api/axiosInstance";

export const getAllRoutes  = ()      =>
  api.get("/routes");

export const getRouteById = (id)    =>
  api.get(`/routes/${id}`);

export const addRoute     = (data)  =>
  api.post("/routes", data);

export const updateRoute  = (id, d) =>
  api.put(`/routes/${id}`, d);

export const deleteRoute  = (id)    =>
  api.delete(`/routes/${id}`);

File: BusTicketBookingSystem/src/main/java/com/ash/bbus/web/BusTicketBookingSystem/service/RouteService.java
package com.ash.bbus.web.BusTicketBookingSystem.service;

import com.ash.bbus.web.BusTicketBookingSystem.dto.RouteDTO;
import java.util.List;

public interface RouteService {

    RouteDTO addRoute(RouteDTO dto);
    RouteDTO getRouteById(Long id);
    List<RouteDTO> getAllRoutes();
    RouteDTO updateRoute(Long id, RouteDTO dto);
    void deleteRoute(Long id);
}